src/node_v8.cc

```diff
@@ -389,6 +389,72 @@ void GCProfiler::Stop(const FunctionCallbackInfo<v8::Value>& args) {
                                 .ToLocalChecked());
 }
 
+static void CostBeforeGC(Isolate* isolate,
+                         v8::GCType gc_type,
+                         v8::GCCallbackFlags flags,
+                         void* data) {
+  Environment* env = static_cast<Environment*>(data);
+  if (env->current_gc_type() != 0) {
+    return;
+  }
+  env->set_current_gc_type(gc_type);
+  env->set_gc_start_time(uv_hrtime());
+}
+
+static void CostAfterGC(Isolate* isolate,
+                        v8::GCType gc_type,
+                        v8::GCCallbackFlags flags,
+                        void* data) {
+  Environment* env = static_cast<Environment*>(data);
+  if (env->current_gc_type() != gc_type) {
+    return;
+  }
+  env->set_current_gc_type(0);
+  env->set_gc_time(uv_hrtime() - env->gc_start_time());
```

Comment:
> My idea is we can calculate the gc load from the value returned bygetGCCost, and use GCProfiler to take the GC details if necessary, like cpuUsage and cpuprofile.

If it's modeled after `process.cpuUsage()`, this should be accumulative here (`env->gc_time()` + `uv_hrtime() - env->gc_start_time()`).

Reply:
Oh! `env->gc_time() + uv_hrtime() - env->gc_start_time()` is right.